services/sonar/sonar-base.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const Joi = require('joi')
+const serverSecrets = require('../../lib/server-secrets')
+const { BaseJsonService } = require('..')
+const { isLegacyVersion } = require('./sonar-helpers')
+
+const schema = Joi.object({
+  component: Joi.object({
+    measures: Joi.array()
+      .items(
+        Joi.object({
+          metric: Joi.string(),
```

Comment:
Should this be `.required()`

Reply:
I want to say I ran into an issue with the API response occasionally had the measures array that would have one or two items that did not have a `metric` property, resulting in `invalid response data`. It's been a while though so I'll double check

Reply:
Have made this required, and no issues 👍